pkg/tracing/tracing.go

```diff
@@ -0,0 +1,58 @@
+package tracing
+
+import (
+	"context"
+	"os"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	sdktrace "go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.7.0"
+	"google.golang.org/grpc"
+)
+
+// SetupTracer is a helper method for registering and starting a new TracerProvider and Propagator.
+// The returned TracerProvider should be shutdown with .Shutdown()
+func SetupTracer(ctx context.Context, opts ...sdktrace.TracerProviderOption) (*sdktrace.TracerProvider, error) {
+	tp := sdktrace.NewTracerProvider(opts...)
+	otel.SetTracerProvider(tp)
+
+	prp := propagation.NewCompositeTextMapPropagator(
+		propagation.TraceContext{},
+		propagation.Baggage{},
+	)
+	otel.SetTextMapPropagator(prp)
+
+	return tp, nil
+}
+
+// NewOTLPExporter returns an OTEL compatible exporter given a grpc endpoint
+func NewOTLPExporter(ctx context.Context, endpoint string, dialOpts ...grpc.DialOption) (*otlptrace.Exporter, error) {
+	conn, err := grpc.DialContext(ctx, endpoint, dialOpts...)
```

Comment:
I'm wondering whether we should treat `endpoint` here the same way we do with `GRPCOptions` and call that here to get our additional "standard" options, WDYT?